File: Dynamic Programming/Knapsack/recursive.cpp
class Solution {
    public:
      int solve(int i, int capacity, vector<int> &val, vector<int> &wt){
          if(capacity==0) return 0;
          if(i>=wt.size()) return 0;
          int take=0;
          if(capacity >= wt[i])
              take = val[i] + solve(i+1, capacity-wt[i], val, wt);
          int skip = solve(i+1, capacity, val, wt);
          return max(take, skip);
      }
      
      int knapSack(int capacity, vector<int> &val, vector<int> &wt) {
          return solve(0, capacity, val, wt);
      }
  };

File: Dynamic Programming/DP on Subsequences/Unbounded Knapsack/recursive.java
class recursive {

    private int helper(int n, int capacity, int val[], int wt[], int i) {
        if (i >= n) return 0;
        if (capacity <= 0) return 0;

        int take = 0;
        int notTake = helper(n, capacity, val, wt, i + 1);

        if (capacity - wt[i] >= 0) {
            take = val[i] + helper(n, capacity - wt[i], val, wt, i);
        }

        return Math.max(take, notTake);
    }

    public int knapSack(int val[], int wt[], int capacity) {
        int n = val.length;
        return helper(n, capacity, val, wt, 0);
    }

}
